### DISCUSSION TITLE
Support for e.g. `dtNavMeshQuery::findPolysAroundCircle`

### DISCUSSION THREAD

**Discussion by rob-myers** [isaac-mason/recast-navigation-js#298] [Ideas]
Is there any way to currently access such methods?

If not, what would I need to do?

Maybe extend `NavMeshQuery.h`, compile the WASM, provide JS wrappers, and extend the types?

**Comment by isaac-mason** [comment]
Hi @rob-myers,

Not all dtNavMeshQuery functionality is exposed yet. You're pretty much spot on re the process to add this method.

1) Extend NavMeshQuery.h: https://github.com/isaac-mason/recast-navigation-js/blob/5c7fe08bd4d73519716b7c5743f3e3960b30ab85/packages/recast-navigation-wasm/src/NavMeshQuery.h

2) Add the new methods to recast-navigation.idl: https://github.com/isaac-mason/recast-navigation-js/blob/5c7fe08bd4d73519716b7c5743f3e3960b30ab85/packages/recast-navigation-wasm/recast-navigation.idl#L449

3) Building @recast-navigation/wasm will auto-generate typescript types using `recast-navigation.idl`

4) Extend the NavMeshQuery js wrapper in @recast-navigation/core: https://github.com/isaac-mason/recast-navigation-js/blob/main/packages/recast-navigation-core/src/nav-mesh-query.ts#L47

If you're interested in giving this a go and submitting a PR, I'd be very happy to help! I've also got some development docs here for setting up your environment: https://github.com/isaac-mason/recast-navigation-js/blob/main/DEVELOPMENT.md

**Reply by rob-myers** [reply]
Hey, thanks for the info.

I created a PR which adds two functions and includes a story.
> https://github.com/isaac-mason/recast-navigation-js/pull/300

**Reply by isaac-mason** [reply]
Great! I'll take a look

**Reply by isaac-mason** [reply]
Your PR has been merged and released, thanks!
